src/riak_kv_eleveldb_backend.erl

```diff
@@ -464,18 +465,18 @@ range_scan(FoldIndexFun, Buffer, Opts, #state{fold_opts=_FoldOpts,
         end,
     StartK2 = [{Field, Val} || {Field, _Type, Val} <- StartK],
     StartK3 = riak_ql_ddl:make_key(Mod, LK, StartK2),
-    StartKey = eleveldb_ts:encode_key(StartK3),
-    StartKey2 = to_object_key(Bucket, StartKey),
+    StartK4 = list_to_tuple([Val || {_Type, Val} <- StartK3]), %% TODO: Avoid adding/removing type info
```

Comment:
this `list_to_tuple([V || {_T, V} <- Key])` should really be wrapped in a function somewhere that's well named and has some edoc about what the heck it's doing and why... it's pretty cryptic trying to understand it reading through the code.